src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -0,0 +1,293 @@
+         ctx.conv_problem.GetOutDataType() == miopenInt8))
+        return false;
+    if(!ctx.direction.IsForward())
+        return false;
+    if(!ctx.Is2d())
+        return false;
+    if(ctx.GetStream().GetDeviceName() != "gfx908")
+        return false;
+    if(!ctx.IsLayoutNHWC())
+        return false;
+
+    std::vector<DeviceConvFwdPtr_t> conv_ptrs;
+    add_device_conv2d_fwd_xdl_nhwc_kyxc_nhwk_int8_instances_t(conv_ptrs);
+    assert(!conv_ptrs.empty());
+    const auto args = CKArgs{ctx};
+    if(!std::all_of(args.strides.begin(), args.strides.end(), [&](auto x) { return x == 1; }))
+        return false;
+
+    for(auto& conv_ptr : conv_ptrs)
+    {
+        auto argument_ptr = conv_ptr.MakeArgumentPointer(nullptr,
+                                                         nullptr,
+                                                         nullptr,
+                                                         args.N,
+                                                         args.K,
+                                                         args.C,
+                                                         args.input,
+                                                         args.filter,
+                                                         args.input,
+                                                         args.strides,
+                                                         args.dilation,
+                                                         args.lPadding,
+                                                         args.rPadding);
+        if(conv_ptr.IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
+#endif
+}
+
+ConvSolution ConvHipImplicitGemmFwdXdlops::GetSolution(
+    const ConvolutionContext& ctx, const PerformanceConfigHipImplicitGemmFwdXdlops& config) const
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = ctx;
+    std::ignore = config;
+    return {};
+#else
+    ConvSolution result;
+    const auto args        = CKArgs{ctx};
+    result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            std::vector<DeviceConvFwdPtr_t> conv_ptrs;
+            add_device_conv2d_fwd_xdl_nhwc_kyxc_nhwk_int8_instances_t(conv_ptrs);
+            auto& conv_ptr       = conv_ptrs.at(config.index);
+            const auto& data_ctx = primitive_parameters.CastTo<conv::DataInvokeParams>();
+            const auto& tensors  = data_ctx.tensors;
+            auto argument_ptr    = conv_ptr.MakeArgumentPointer(
+                const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+                    static_cast<const void*>(tensors.in)),
+                const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+                    static_cast<const void*>(tensors.w)),
```

Comment:
If removal of constness is necessary, then there is a flaw in design (either in the core of the library or in CK)

@DrizztDoUrden can you have a look at this specific place?